### PR TITLE
Kzm ncm727 setup password

### PR DESCRIPTION
**Pull request by 3mdeb-karolzmyslowski**
Add Setup Password option

### CODE REVIEW

File: .config [hunk 1]
```diff
@@ -0,0 +1,906 @@
+#
```
Comment (macpijan) [inline .config:1]: @3mdeb-karolzmyslowski .config should never be commited. How did you do that anyway? It should be in the `.gitignore`.

File: payloads/external/Makefile.inc [hunk 2]
```diff
@@ -202,6 +202,8 @@ $(obj)/UEFIPAYLOAD.fd edk2: $(DOTCONFIG) $(IPXE_EFI)
 		CONFIG_EDK2_DASHARO_INTEL_ME_OPTIONS=$(CONFIG_EDK2_DASHARO_INTEL_ME_OPTIONS) \
 		CONFIG_EDK2_DASHARO_USB_CONFIG=$(CONFIG_EDK2_DASHARO_USB_CONFIG) \
 		CONFIG_EDK2_DASHARO_NETWORK_CONFIG=$(CONFIG_EDK2_DASHARO_NETWORK_CONFIG) \
+		CONFIG_EDK2_DASHARO_NETWORK_BOOT_DEFAULT_ENABLE=$(CONFIG_EDK2_DASHARO_NETWORK_BOOT_DEFAULT_ENABLE) \
```
Comment (macpijan) [inline payloads/external/Makefile.inc:205]: @3mdeb-karolzmyslowski About this commit message:

`Makefile.inc, Kconfig, Makefile: Disable network boot, added .config `

Since it simply adds network boot enable option into edk2, I would name it:

`payloads/external/edk2: introduce DASHARO_NETWORK_BOOT_DEFAULT_ENABLE`, or similar.

File: payloads/external/edk2/Kconfig [hunk 3]
```diff
@@ -282,6 +282,18 @@ config EDK2_DASHARO_NETWORK_CONFIG
 	help
 	  Enables the EDK2 network stack configuration submenu in Dasharo System Features.
 
+config EDK2_DASHARO_NETWORK_BOOT_DEFAULT_ENABLE
+	bool "Enable EDK2 network boot option by default"
+	default n
+	help
+	  Enables the network boot option in EDK2 by default.
+
+config EDK2_SETUP_PASSWORD
```
Comment (macpijan) [inline payloads/external/edk2/Kconfig:291]: @3mdeb-karolzmyslowski 

About this commit name:

`.config, Makefile.inc, Kconfig, Makefile: Add setup password `

If you look at the changes, it should go into two separate commits. One of them adds new option to edk2 payload, and is independend of the hardware platform.

The other commit should simply enable this option in certain NovaCustom configs.

**Review by macpijan — CHANGES_REQUESTED** [review]

File: configs/config.novacustom_nv4x_adl [hunk 4]
```diff
@@ -1,50 +1,912 @@
+#
```
Comment (miczyg1) [inline configs/config.novacustom_nv4x_adl:1]: The proper way to commit a board config is to do:

```
make savedefconfig
cp defconfig configs/config.novacustom_nv4x_adl
git add configs/config.novacustom_nv4x_adl
git commit ...
```

File: payloads/external/Makefile.inc (hunk 2 above)
Comment (3mdeb-karolzmyslowski) [reply payloads/external/Makefile.inc:205]: OK, but how I could correct it? By option `--amend`? By squashing?

**Comment by 3mdeb-karolzmyslowski** [comment]
I've seen this error log (https://github.com/Dasharo/coreboot/actions/runs/4355342220/jobs/7611828161). 
```
    LD            futility/futility
/usr/bin/ld: /__w/coreboot/coreboot/build/util/vboot_lib/libvboot_util.a(2crypto.o): warning: relocation in read-only section `.rodata.vb2_sig_names'
/usr/bin/ld: warning: creating DT_TEXTREL in a PIE
    CREATE GBB (without BMPFV)
successfully created new GBB to: build/gbb.stub.tmp
    SETUP GBB
/__w/coreboot/coreboot/util/chromeos/gen_test_hwid.sh: 16: hexdump: not found
 - import root_key from 3rdparty/vboot/tests/devkeys/root_key.vbpubk: success
 - import recovery_key from 3rdparty/vboot/tests/devkeys/recovery_key.vbpubk: success
successfully saved new image to: build/gbb.region.tmp
    WRITE GBB
E: File to add is 65 bytes and would overflow 64-byte target region
E: Failed while operating on 'RO_FRID' region!
E: The image will be left unmodified.
make: *** [src/security/vboot/Makefile.inc:278: build_complete] Error 1
Error: Process completed with exit code 2.
```
Problem is with boundary of the region (RO_FRID).

**Comment by miczyg1** [comment]
> I've seen this error log (https://github.com/Dasharo/coreboot/actions/runs/4355342220/jobs/7611828161).
> 
> ```
>     LD            futility/futility
> /usr/bin/ld: /__w/coreboot/coreboot/build/util/vboot_lib/libvboot_util.a(2crypto.o): warning: relocation in read-only section `.rodata.vb2_sig_names'
> /usr/bin/ld: warning: creating DT_TEXTREL in a PIE
>     CREATE GBB (without BMPFV)
> successfully created new GBB to: build/gbb.stub.tmp
>     SETUP GBB
> /__w/coreboot/coreboot/util/chromeos/gen_test_hwid.sh: 16: hexdump: not found
>  - import root_key from 3rdparty/vboot/tests/devkeys/root_key.vbpubk: success
>  - import recovery_key from 3rdparty/vboot/tests/devkeys/recovery_key.vbpubk: success
> successfully saved new image to: build/gbb.region.tmp
>     WRITE GBB
> E: File to add is 65 bytes and would overflow 64-byte target region
> E: Failed while operating on 'RO_FRID' region!
> E: The image will be left unmodified.
> make: *** [src/security/vboot/Makefile.inc:278: build_complete] Error 1
> Error: Process completed with exit code 2.
> ```
> 
> Problem is with boundary of the region (RO_FRID).

Yup, if you have a dirty repo, the firmware ID that is injected into RO_FRID region becomes too long and doesn't fit. AS a workaround you may make RO_FRID region 0x100 in size.

**Comment by macpijan** [comment]
@miczyg1 What do you mean by dirty repo? DIrty in a git way?

How it can be dirty in CI, when the previous jobs has passed?

**Comment by miczyg1** [comment]
> @miczyg1 What do you mean by dirty repo? DIrty in a git way?
> 
> How it can be dirty in CI, when the previous jobs have passed?

Dirty in a git way.

Ahh okay it is on CI....

File: .config (hunk 1 above)
Comment (3mdeb-karolzmyslowski) [reply .config]: (Corrected, in `https://github.com/Dasharo/coreboot/pull/309/commits/a5dc727a9bd297e534f3e646375a4c5e2ebe70b4`)

**Comment by 3mdeb-karolzmyslowski** [comment]
@macpijan 
Here, above is log of CI task's execution. Problem is cancelling the compilation process:
```
(...)
Building ... /__w/coreboot/coreboot/payloads/external/edk2/workspace/Dasharo/MdeModulePkg/Universal/Acpi/BootGraphicsResourceTableDxe/BootGraphicsResourceTableDxe.inf [X64]
Building ... /__w/coreboot/coreboot/payloads/external/edk2/workspace/Dasharo/UefiCpuPkg/CpuIo2Dxe/CpuIo2Dxe.inf [X64]
Building ... /__w/coreboot/coreboot/payloads/external/edk2/workspace/Dasharo/MdeModulePkg/Bus/Pci/EhciDxe/EhciDxe.inf [X64]
Building ... /__w/coreboot/coreboot/payloads/external/edk2/workspace/Dasharo/MdeModulePkg/Universal/SerialDxe/SerialDxe.inf [X64]
Building ... /__w/coreboot/coreboot/payloads/external/edk2/workspace/Dasharo/MdeModulePkg/Universal/SecurityStubDxe/SecurityStubDxe.inf [X64]
Error: The operation was canceled.
```
I've got the information that there is problem with the timeout. Could You solve it?

**Comment by macpijan** [comment]
@3mdeb-karolzmyslowski I do not know where does this log come from.
Please link to CI log when yo urefer to it.  In the [latest log](https://github.com/Dasharo/coreboot/actions/runs/4364737576/jobs/7632425442#step:5:3310) I can see different type of error, that perhaps you could solve:

```
     origin/kzm_boot_enable_option is not a valid git reference
```

BTW : https://english.stackexchange.com/questions/30185/you-versus-you-as-polite-form-of-writing

**Comment by 3mdeb-karolzmyslowski** [comment]
@macpijan I see, I had left my test edk2 branch. I correct it in next commit.

**Comment by 3mdeb-karolzmyslowski** [comment]
@macpijan please, check my own squash of the recent changes.

**Review by macpijan — CHANGES_REQUESTED** [review]

**Comment by macpijan** [comment]
@3mdeb-karolzmyslowski You added empty commit: https://github.com/Dasharo/coreboot/pull/309/commits/d20e1183730ac78c31ed5e7e8a1d56d1501bfef4 - what is the point with that?

File: configs/config.novacustom_nv4x_adl [hunk 5]
```diff
@@ -36,15 +39,12 @@ CONFIG_EDK2_SECURE_BOOT=y
...
+CONFIG_EDK2_DASHARO_NETWORK_CONFIG=y
 CONFIG_EDK2_DASHARO_NETWORK_BOOT_DEFAULT_ENABLE=y
+CONFIG_EDK2_SETUP_PASSWORD=y
 CONFIG_EDK2_BOOT_MENU_KEY=0x0011
 CONFIG_EDK2_SETUP_MENU_KEY=0x000C
 # CONFIG_EDK2_SECURE_BOOT_DEFAULT_ENABLE is not set
 CONFIG_PXE_ADD_SCRIPT=y
 CONFIG_PXE_SCRIPT="3rdparty/dasharo-blobs/dasharo/dasharo.ipxe"
 CONFIG_PXE_CUSTOM_BUILD_ID="0123456789"
-CONFIG_ENABLE_EARLY_DMA_PROTECTION=y
```
[the remark this answers is not in the document]
Comment (3mdeb-karolzmyslowski) [reply configs/config.novacustom_nv4x_adl:48]: I don't have intention to remove it, probably problem during rebase.

**Comment by 3mdeb-karolzmyslowski** [comment]
@macpijan 
> you added empty commit: https://github.com/Dasharo/coreboot/commit/d20e1183730ac78c31ed5e7e8a1d56d1501bfef4 - what is the point with that?
My intention was cherry-picking Michal's commits. Apparently, one of them didn't applied.

**Comment by macpijan** [comment]
> My intention was cherry-picking Michal's commits. Apparently, one of them didn't applied.

It was empty (since the changes included in this commir are already there in this branch). Git automatically does not pick empty commits (as it makes very little sense to add commits with 0 content), unless you forces it to do so.

This MR can be replaced with: https://github.com/Dasharo/coreboot/pull/320/files
